### PR TITLE
[tensorflow][build][test] Tensorflow serving 2.11 release changes

### PR DESCRIPTION
**Pull request by hballuru**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Tensorflow 2.11 Inference Release

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [x] Revision B: `sagemaker_remote_tests = "rc"`
- [x] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/inference/docker/2.11/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+FROM base_image AS ec2
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+ARG PYTHON=python3.9
+ARG PYTHON_PIP=python3-pip
+ARG PIP=pip3
+ARG PYTHON_VERSION=3.9.10
+ARG TFS_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow_serving/r2.11_aws_rc1/cpu/2022-10-27-02-08/tensorflow_model_server
```
Comment (shantanutrip) [inline tensorflow/inference/docker/2.11/py3/Dockerfile.cpu:36]: To be updated with the new release

File: tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+FROM base_image AS ec2
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+ARG PYTHON=python3.9
+ARG PYTHON_PIP=python3-pip
+ARG PIP=pip3
+ARG PYTHON_VERSION=3.9.10
+ARG TFS_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow_serving/r2.11_aws_rc1/gpu/2022-10-27-02-50/tensorflow_model_server
```
Comment (shantanutrip) [inline tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu:36]: To be updated with the new release

File: tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.11/license.txt -o /license.txt
```
Comment (shantanutrip) [inline tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu:180]: Thanks for changing this. Instead of writing `tensorflow-2.11` , shall we try to use some `ARG` for eg. `TFS_SHORT_VERSION` and make it `tensorflow-${TFS_SHORT_VERSION}` to prevent any misses in the future?

File: dlc_developer_config.toml [hunk 4]
```diff
@@ -23,9 +23,9 @@ benchmark_mode = false
...
 # Frameworks for which you want to disable both builds and tests
 # available frameworks - ["autogluon", "huggingface_tensorflow", "huggingface_pytorch", "huggingface_tensorflow_trcomp", "huggingface_pytorch_trcomp", "tensorflow", "mxnet", "pytorch"]
-skip_frameworks = []
+skip_frameworks = ["autogluon", "huggingface_tensorflow", "huggingface_pytorch", "huggingface_tensorflow_trcomp", "huggingface_pytorch_trcomp", "mxnet", "pytorch"]
 # Set to false in order to remove datetime tag on PR builds
-datetime_tag = true
+datetime_tag = false
 # Note: Need to build the images at least once with datetime_tag = false
 # before disabling new builds, or tests will fail
 do_build = true
```
Comment (shantanutrip) [inline dlc_developer_config.toml:31]: We can run the following set of tests in first iteration
1. `sagemaker_local_tests = true`
2. `sagemaker_remote_tests = standard`

File: tensorflow/inference/docker/2.11/py3/Dockerfile.cpu [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+ && echo 'deb http://nginx.org/packages/ubuntu/ focal nginx' >> /etc/apt/sources.list \
+ && apt-get update \
+ && apt-get -y install --no-install-recommends \
+    nginx \
+    nginx-module-njs \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN ${PIP} install --no-cache-dir \
+    falcon==3.1.0 \
```
Comment (ohadkatz) [inline tensorflow/inference/docker/2.11/py3/Dockerfile.cpu:185]: Is there a reason these packages need to be pinned?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 6]
```diff
@@ -465,6 +466,7 @@ def _run_dependency_check_test(image, ec2_connection):
             "2.8": ["cpu", "gpu", "hpu"],
             "2.9": ["cpu", "gpu", "hpu"],
             "2.10": ["cpu", "gpu", "hpu"],
+            "2.11": ["cpu","gpu"],
```
Comment (ohadkatz) [inline test/dlc_tests/sanity/test_pre_release.py:469]: nit: space

Comment (hballuru) [reply test/dlc_tests/sanity/test_pre_release.py]: done

File: tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+ARG PYTHON_VERSION=3.9.10
+ARG TFS_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow_serving/r2.11_aws/gpu/2022-11-04-07-27/tensorflow_model_server
+
+# ENV variable to be pased to SageMaker stage
+ENV PIP=${PIP}
+ENV PYTHON=${PYTHON}
+
+
+ENV NCCL_VERSION=2.8.4-1+cuda11.2
+ENV CUDNN_VERSION=8.1.0.77-1+cuda11.2
```
[the remark this answers is not in the document]
Comment (shantanutrip) [reply tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu]: Done

File: test/test_utils/__init__.py [hunk 8]
```diff
@@ -52,8 +52,8 @@ def get_ami_id_ssm(region_name, parameter_path):
     ami_id = eval(ami['Parameter']['Value'])['image_id']
     return ami_id
 
-UBUNTU_18_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning Base AMI (Ubuntu 18.04) Version ??.?")
-UBUNTU_18_BASE_DLAMI_US_EAST_1 = get_ami_id_boto3(region_name="us-east-1", ami_name_pattern="Deep Learning Base AMI (Ubuntu 18.04) Version ??.?")
+UBUNTU_18_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning AMI GPU CUDA 11.1.1 (Ubuntu 18.04) ????????")
```
[the remark this answers is not in the document]
Comment (shantanutrip) [reply test/test_utils/__init__.py]: Done

File: tensorflow/inference/docker/2.11/py3/Dockerfile.cpu (hunk 5 above)
Comment (shantanutrip) [reply tensorflow/inference/docker/2.11/py3/Dockerfile.cpu:185]: We have been using the same `falcon` versions for `TF2.10,2.9 and 2.8`. So, we might want to pin these for this release in the same way.

File: tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-${TFS_SHORT_VERSION}/license.txt -o /license.txt
```
Comment (ohadkatz) [inline tensorflow/inference/docker/2.11/py3/cu112/Dockerfile.gpu:182]: same question with existence of license file, and if this needs to be separate for tfs and tf

File: tensorflow/inference/docker/2.11/py3/Dockerfile.cpu [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+RUN HOME_DIR=/root \
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance*
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-${TFS_SHORT_VERSION}/license.txt -o /license.txt
```
[the remark this answers is not in the document]
Comment (shantanutrip) [reply tensorflow/inference/docker/2.11/py3/Dockerfile.cpu:141]: I think that the Dockerfiles have the same license for training and inference images. I don't think we need separate license files.

File: dlc_developer_config.toml [hunk 11]
```diff
@@ -46,14 +46,14 @@ ec2_tests = true
 
 ### SM specific tests
 ### Off by default
-sagemaker_local_tests = false
+sagemaker_local_tests = true
```
Comment (dkey-amazon) [inline dlc_developer_config.toml:49]: Not a big deal, but the check box for this in the PR description is not checked.

File: dlc_developer_config.toml [hunk 12]
```diff
@@ -46,14 +46,14 @@ ec2_tests = true
...
 
 # SM remote test valid values:
 # "off" --> do not trigger sagemaker remote tests (default)
 # "standard" --> run standard sagemaker remote tests from test/sagemaker_tests
 # "rc" --> run release_candidate_integration tests
 # "efa" --> run efa sagemaker tests
-sagemaker_remote_tests = "off"
+sagemaker_remote_tests = "efa"
```
Comment (dkey-amazon) [inline dlc_developer_config.toml:56]: Again, the check box for this in the PR description is not checked.

**Review by dkey-amazon — COMMENTED** [review]
A few minor points to correct but overall looks good.

File: dlc_developer_config.toml [hunk 13]
```diff
@@ -23,17 +23,17 @@ huggingface_trcomp_mode = false
 trcomp_mode = false
 # Please only set it to True if you are preparing a Benchmark related PR
 # Do remember to revert it back to False before merging any PR (including Benchmark dedicated PR)
-benchmark_mode = false
+benchmark_mode = true
```
[the remark this answers is not in the document]
Comment (hballuru) [reply dlc_developer_config.toml]: Tests have passed successfully for the below commits:

[091477d](https://github.com/aws/deep-learning-containers/pull/2372/commits/091477d4758547770651d1f40f7e6d78e20035c3) benchmark false, sm remote tests standard, do_build true
[14cf637](https://github.com/aws/deep-learning-containers/pull/2372/commits/14cf637297ac14b7f35bc290a8bfae766b9191f5) benchmark false, sm remote tests standard, do_build false
[9fbd9f6](https://github.com/aws/deep-learning-containers/pull/2372/commits/9fbd9f63af46c6b680ed579919e6d3762b5f1b44) benchmark false, sm remote tests efa, do_build false

[42be88c](https://github.com/aws/deep-learning-containers/pull/2372/commits/42be88c3f027a5c9ddf5296264cb12ae7209cb7b) benchmark true, sm remote tests standard, do_build false
[66029b3](https://github.com/aws/deep-learning-containers/pull/2372/commits/66029b3229350f71096cc51686743dbe9354a895) benchmark true, sm remote tests era, do_build false

[the remark this answers is not in the document]
Comment (shantanutrip) [reply dlc_developer_config.toml]: As stated by Harsha, we have already run tests with all configurations. For eg., we have run tests such as sanity, sagemaker local, sagemaker remote etc.

File: dlc_developer_config.toml (hunk 11 above)
Comment (shantanutrip) [reply dlc_developer_config.toml:49]: I have checked the box now, thanks!!

File: dlc_developer_config.toml (hunk 12 above)
Comment (shantanutrip) [reply dlc_developer_config.toml]: I have checked the box now, thanks!!

File: dlc_developer_config.toml (hunk 13 above)
[the remark this answers is not in the document]
Comment (dkey-amazon) [reply dlc_developer_config.toml]: Ok, cool.  I we should be good here then.

File: tensorflow/inference/docker/2.11/py3/Dockerfile.cpu (hunk 10 above)
[the remark this answers is not in the document]
Comment (dkey-amazon) [reply tensorflow/inference/docker/2.11/py3/Dockerfile.cpu:141]: Just to double check, have you verified that this license file exists for TF 2.11 @shantanutrip?

**Review by dkey-amazon — APPROVED** [review]

**Review by ohadkatz — APPROVED** [review]

**Review by ohadkatz — APPROVED** [review]
LGTM! 🚢

File: tensorflow/inference/docker/2.11/py3/Dockerfile.cpu (hunk 10 above)
[the remark this answers is not in the document]
Comment (ohadkatz) [reply tensorflow/inference/docker/2.11/py3/Dockerfile.cpu:141]: @dkey-amazon I created the license for EC2 👍 .

**Review by ohadkatz — APPROVED** [review]
LGTM! :shipit:

**Review by shantanutrip — APPROVED** [review]
